### PR TITLE
Blog post announcing storing charts in OCI as a full feature

### PR DESCRIPTION
**Pull request by scottrigby**
Good to get out there! Set a target date that feels realistic. Can move if needed.

Thanks @jdolitsky and @mattfarina for collaboration. Thanks @bridgetkromhout for review in HackMD.
For more history, see: https://hackmd.io/3ixHqHw5TyqcY9yXt1DPPA?view

### CODE REVIEW

**Comment by mattfarina** [comment]
Shouldn't be merged until Monday

File: content/en/blog/2022-02-28-storing-charts-in-oci.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+$ helm push demo-0.1.0.tgz oci://r.example.com/myuser
+Pushed: r.example.com/myuser/demo:0.1.0
+Digest: sha256:7ed393daf1ffc94803c08ffcbecb798fa58e786bebffbab02da5458f68d0ecb0
+```
+
+More detail on [working with registries can be found in the Helm documentation](https://helm.sh/docs/topics/registries/).
+
+## The Helm SDK
+
+The Helm SDK, which is useful for those building tools on top of Helm functionality, also includes that ability to work with registries programatically. The following example illustrates pushing a chart to a registry:
```
Comment (sabre1041) [inline content/en/blog/2022-02-28-storing-charts-in-oci.md:45]:
```suggestion
The Helm SDK, which is useful for those building tools to integrate with Helm, also includes support to work with registries programatically. The following example illustrates pushing a chart to a registry:
```

File: content/en/blog/2022-02-28-storing-charts-in-oci.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+More detail can be found in the [documentation for the registry package](https://pkg.go.dev/helm.sh/helm/v3/pkg/registry).
+
+## Limitations
+
+There are some limitations when using registries to store charts compared to Helm repositories or storing container images in registries.
+
+Helm repositores can be added and searched from the local Helm client. This is similar to how repositories work with other package managers such as zypper or apt. When working with OCI registries, this is not an option. OCI based registries don't provide standard APIs to facilitate searching.
+
+While the OCI specification provides for artifacts, not all registries support storing Helm charts or other artifacts that are not container images. Before choosing a registry you need to make sure it supports storing Helm charts.
```
Comment (sabre1041) [inline content/en/blog/2022-02-28-storing-charts-in-oci.md:74]:
```suggestion
While the OCI specification provides support for artifacts, not all registries support storing Helm charts or other artifacts that are not container images. Before choosing a registry, you should confirm whether it supports storing Helm charts.
```

File: content/en/blog/2022-02-28-storing-charts-in-oci.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+## Artifact Hub Support
+
+[Artifact Hub](https://artifacthub.io/), another CNCF project, provides a means to search and discover cloud native assets including charts. Helm charts stored in OCI based registries can be listed on Artifact Hub, which already knows how to work with them. More details on working with Artifact Hub and Helm charts in container registries can be found in their [documentation](https://artifacthub.io/docs/topics/repositories/#helm-charts-repositories).
```
Comment (sabre1041) [inline content/en/blog/2022-02-28-storing-charts-in-oci.md:78]:
```suggestion
[Artifact Hub](https://artifacthub.io/), another CNCF project, provides a means to search and discover cloud native assets, including charts. Helm charts stored in OCI based registries can be listed on Artifact Hub, which already knows how to work with them. More details on working with Artifact Hub and Helm charts in container registries can be found in their [documentation](https://artifacthub.io/docs/topics/repositories/#helm-charts-repositories).
```

File: content/en/blog/2022-02-28-storing-charts-in-oci.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+## ORAS
+
+The [OCI Registry as Storage (ORAS) project](https://oras.land/), another CNCF project, provides the library Helm uses to work with registries. The ORAS project bills itself as:
```
Comment (sabre1041) [inline content/en/blog/2022-02-28-storing-charts-in-oci.md:82]:
```suggestion
The [OCI Registry as Storage (ORAS) project](https://oras.land/), another CNCF project, is used by Helm as the underlying library for working with registries. The ORAS project bills itself as:
```

File: content/en/blog/2022-02-28-storing-charts-in-oci.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+> Registries are evolving as generic artifact stores. To enable this goal, the ORAS project provides a way to push and pull OCI Artifacts to and from OCI Registries.
+
+If you want to work with other artifacts in registries the ORAS project may provide some tools to help you.
```
Comment (sabre1041) [inline content/en/blog/2022-02-28-storing-charts-in-oci.md:86]:
```suggestion
If you want to work with other artifacts stored in registries the ORAS project may provide some tools to help you.
```

File: content/en/blog/2022-02-28-storing-charts-in-oci.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
+---
+title: "Storing Helm Charts in OCI Registries"
+slug: "storing-charts-in-oci"
+authorname: "Scott Rigby, Josh Dolitsky, Matt Farina"
+author: "@scottrigby, @jdolitsky, @mattfarina"
+authorlink: "https://helm.sh"
+date: "2022-02-28"
+---
+
+With the release of Helm 3.8.0, Helm is able to store and work with charts in container registries, as an alternative to [Helm repositories](https://helm.sh/docs/topics/chart_repository/). This feature, which used to be an experimental feature, is now generally available.
```
[the remark this answers is not in the document]
Comment (scottrigby) [reply content/en/blog/2022-02-28-storing-charts-in-oci.md:10]: Not sure at this one. The OCI artifact integration in Helm is only intended to push, pull, install (etc) packaged charts, not other artifacts, right?

[the remark this answers is not in the document]
Comment (scottrigby) [reply content/en/blog/2022-02-28-storing-charts-in-oci.md:10]: Looks like this is a duplicate review comment.

File: content/en/blog/2022-02-28-storing-charts-in-oci.md (hunk 5 above)
Comment (scottrigby) [reply content/en/blog/2022-02-28-storing-charts-in-oci.md:86]: The existing wording covers both storing artifacts in, and retrieving artifacts from registries, right?

To me this change might imply that ORAS only works with other artifacts that are already stored in registries 🤔

File: content/en/blog/2022-02-28-storing-charts-in-oci.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jdolitsky) [reply content/en/blog/2022-02-28-storing-charts-in-oci.md:10]: I agree with scott here. Helm does not push/pull images

**Review by jdolitsky — APPROVED** [review]
Just updated the SDK section. LGTM!

**Comment by scottrigby** [comment]
@jdolitsky SDK example update look great! appreciate your `check()` code reduction too – while not always, in this case it really does make the example read more simply 👀 🤌 🤏

OK thanks all, posting now 💯 🤝

File: content/en/blog/2022-02-28-storing-charts-in-oci.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (sabre1041) [reply content/en/blog/2022-02-28-storing-charts-in-oci.md:10]: Yeah looking back, I'm ok with the original version

**Pull request merged**